### PR TITLE
[libmysql/libmariadb] Export unofficial cmake targets

### PR DESCRIPTION
**Pull request by JackBoosY**
In order to use libmysql and libmariadb easily, export the unofficial targets.
For libmysql, since the library and the target generated by its dynamic build and static build is inconsistent, a wrapper is added to unify the output interface.

When the executable in the bin is cleaned up, the executable related pdb also need to be cleaned up.

This PR prepares for poco update.

### CODE REVIEW

File: ports/libmysql/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,7 @@
 vcpkg_fail_port_install(ON_TARGET "UWP" ON_ARCH "x86")
 
+vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)
```
Comment (JackBoosY) [inline ports/libmysql/portfile.cmake:3]: In _MYSQL_SOURCE\libmysql\CMakeLists.txt_ line 285:
```
# Merge several convenience libraries into one big mysqlclient
# and link them together into shared library.
MERGE_LIBRARIES_SHARED(libmysql ${LIBS_TO_MERGE}
  EXPORTS
  ${CLIENT_API_FUNCTIONS}
  ${CLIENT_API_FUNCTIONS_UNDOCUMENTED}
  ${CLIENT_API_NONBLOCKING_FUNCTIONS}
  COMPONENT SharedLibraries)
```

This port should always be dynamic.

**Review by PhoebeHui — APPROVED** [review]

File: ports/libmysql/CONTROL [hunk 2]
```diff
@@ -1,6 +1,7 @@
 Source: libmysql
 Version: 8.0.20
+Port-Version: 1
 Homepage: https://github.com/mysql/mysql-server
 Build-Depends: boost-algorithm, boost-geometry, boost-optional, boost-functional, boost-graph, openssl, icu, libevent, lz4, zlib
 Description: A MySQL client library for C development.
-Supports: !(windows&x86)&!uwp
\ No newline at end of file
+Supports: !(windows&(x86))&!uwp
```
Comment (JackBoosY) [inline ports/libmysql/CONTROL:7]:
```suggestion
Supports: !(windows&x86)&!uwp
```

[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/libmysql/CONTROL]: Sorry,  I forgot  to revert it.  Now it's ok.

**Comment by BillyONeal** [comment]
I'm not ignoring this one but I want to go over it with some other team members who understand the implications better than I do; should land Monday at the latest.

Thanks for your contribution!

**Comment by JackBoosY** [comment]
Done, already tested usage with poco.

**Comment by JackBoosY** [comment]
Depends on #14396.

**Review by PhoebeHui — APPROVED** [review]

File: ports/libmariadb/export-cmake-targets.patch [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+         ARCHIVE DESTINATION lib)
+ 
++install(EXPORT unofficial-libmariadb-targets DESTINATION share/unofficial-libmariadb)
++
++file(WRITE "${CMAKE_CURRENT_BINARY_DIR}/unofficial-libmariadb-config.cmake"
++"include(CMakeFindDependencyMacro)
++find_dependency(ZLIB)
++find_dependency(OpenSSL)
++include(\"\${CMAKE_CURRENT_LIST_DIR}/unofficial-libmariadb-targets.cmake\")
++")
```
Comment (strega-nil) [inline ports/libmariadb/export-cmake-targets.patch:21]:
```suggestion
+[[include(CMakeFindDependencyMacro)
+find_dependency(ZLIB)
+find_dependency(OpenSSL)
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-libmariadb-targets.cmake")
+]])
```

**Comment by strega-nil** [comment]
Merging after a run-through of CI.

**Pull request merged**